pattern_import_export_xlsx/models/ir_exports.py

```diff
@@ -17,30 +17,68 @@ class IrExports(models.Model):
     export_format = fields.Selection(selection_add=[("xlsx", "Excel")])
 
     @api.multi
-    def _create_xlsx_file(self):
+    def _create_xlsx_file(self, records):
         self.ensure_one()
-        pattern_file = BytesIO()
-        book = xlsxwriter.Workbook(pattern_file)
-        sheet = book.add_worksheet(self.name)
-        cell_style = book.add_format({"bold": True})
-        ad_sheet_list = {}
-        for col, header in enumerate(self._get_header()):
-            sheet.write(0, col, header, cell_style)
-        # Manage others tab of Excel file!
-        for select_tab in self._get_select_tab():
-            select_tab_name = select_tab.name
-            field_name = select_tab.field_id.name
-            ad_sheet_name = select_tab_name + " (" + field_name + ")"
-            if ad_sheet_name not in ad_sheet_list:
-                ad_sheet, ad_row = select_tab._generate_additional_sheet(
-                    book, cell_style
-                )
-                ad_sheet_list[ad_sheet.name] = (ad_sheet, ad_row)
-            else:
-                ad_sheet = ad_sheet_list[ad_sheet.name][0]
-                ad_row = ad_sheet_list[ad_sheet.name][1]
-            select_tab._add_xlsx_constraint(sheet, col, ad_sheet, ad_row)
-        return book, sheet, pattern_file
+        book = openpyxl.Workbook()
+        main_sheet = self._build_main_sheet_structure(book)
+        self._populate_main_sheet_rows(main_sheet, records)
+        tab_data = self.export_fields._get_tab_data()
+        self._create_tabs(book, tab_data)
+        main_sheet_length = len(records.ids) + 1
+        self._create_validators(main_sheet, main_sheet_length, tab_data)
+        book.close()
+        xlsx_file = BytesIO()
+        book.save(xlsx_file)
+        return xlsx_file
+
+    def _build_main_sheet_structure(self, book):
+        """
+        Write main sheet header and other style details
+        """
+        main_sheet = book["Sheet"]
+        main_sheet.title = self.name
+        for col, header in enumerate(self._get_header(), start=1):
+            main_sheet.cell(row=1, column=col, value=header)
+        return main_sheet
+
+    def _populate_main_sheet_rows(self, main_sheet, records):
+        """
+        Get the actual data and write it row by row on the main sheet
+        """
+        for row, values in enumerate(self._get_data_to_export(records), start=2):
+            for col, header in enumerate(self._get_header(), start=1):
```

Comment:
maybe we should put the result of self._get_header() into a variable to avoid to call the method for each row